Frameworks/UIKit/UIView.mm

```diff
@@ -198,27 +199,68 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
         }
     }
 
-    // gesture priority list
-    const static id s_gesturesPriority[] = {[UIPinchGestureRecognizer class],
-                                            [UISwipeGestureRecognizer class],
-                                            [UIPanGestureRecognizer class],
-                                            [UILongPressGestureRecognizer class],
-                                            [UITapGestureRecognizer class] };
 
-    const static int s_numGestureTypes = sizeof(s_gesturesPriority) / sizeof(s_gesturesPriority[0]);
+    //
+    // Remove any gestures that should not simulatanously be recognized
+    //
 
-    //  Process all gestures
-    for (int i = 0; i < s_numGestureTypes; i++) {
-        id curgestureClass = s_gesturesPriority[i];
-        id gestures = [g_curGesturesDict objectForKey:curgestureClass];
-        if ([curgestureClass _fireGestures:gestures]) {
-            handled = true;
-            if (handled && DEBUG_GESTURES) {
-                TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
+    // First find the active gesture
+    UIGestureRecognizer* activeGesture = nil;
+
+    int count = [g_currentlyTrackingGesturesList count];
+    for (int i = count - 1; i >= 0; i--) {
```

Comment:
@aballway I agree that for in is quicker to read so I will make the changes.  Just curious as to what your team prefers given the mix in this file.  I will make the changes.